src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3725,3 +3726,141 @@ mongoc_collection_watch (const mongoc_collection_t *coll,
+      server_stream =
+         mongoc_cluster_stream_for_writes (&collection->client->cluster,
+                                           NULL /* mongoc_client_session_t */,
+                                           reply_ptr,
+                                           error);
+      if (server_stream->sd->max_wire_version < WIRE_VERSION_4_4) {
+         // Raise an error required by the specification:
+         // "Drivers MUST manually raise an error if this option is specified
+         // when creating an index on a pre 4.4 server."
+         bson_set_error (
+            error,
+            MONGOC_ERROR_COMMAND,
+            MONGOC_ERROR_PROTOCOL_BAD_WIRE_VERSION,
+            "The selected server does not support the commitQuorum option");
+         GOTO (fail);
+      }
+   }
+
+   // Build the createIndexes command.
+   BSON_ASSERT (
+      BSON_APPEND_UTF8 (&cmd, "createIndexes", collection->collection));
+   bson_t indexes;
+   BSON_ASSERT (BSON_APPEND_ARRAY_BEGIN (&cmd, "indexes", &indexes));
+   for (uint32_t idx = 0; idx < n_models; idx++) {
+      /*
+         Append a document of this form:
+         <idx>: {
+             key: {
+                 <key-value_pair>,
+                 <key-value_pair>,
+                 ...
+             },
+             name: <index_name>,
+             <option1>,
+             <option2>,
+             ...
+         },
+      */
+
+      const char *idx_str_ptr;
+      char idx_str[16];
+      BSON_ASSERT (
+         bson_uint32_to_string (idx, &idx_str_ptr, idx_str, sizeof (idx_str)) <=
+         sizeof (idx_str));
+      bson_t index;
+      BSON_ASSERT (BSON_APPEND_DOCUMENT_BEGIN (&indexes, idx_str_ptr, &index));
+      BSON_ASSERT (BSON_APPEND_DOCUMENT (&index, "key", models[idx]->keys));
+      bson_iter_t name_iter;
+      if (models[idx]->opts &&
+          bson_iter_init_find (&name_iter, models[idx]->opts, "name")) {
+         // `name` was specified as an index option.
+      } else {
+         // No `name` was specified. Create index `name` from keys.
+         char *name =
+            mongoc_collection_keys_to_index_string (models[idx]->keys);
+         BSON_ASSERT (name);
+         BSON_ASSERT (BSON_APPEND_UTF8 (&index, "name", name));
+         bson_free (name);
+      }
+
+      if (models[idx]->opts) {
+         BSON_ASSERT (bson_concat (&index, models[idx]->opts));
+      }
+      BSON_ASSERT (bson_append_document_end (&indexes, &index));
+   }
+   BSON_ASSERT (bson_append_array_end (&cmd, &indexes));
+
+   if (!mongoc_client_command_with_opts (collection->client,
+                                         collection->db,
+                                         &cmd,
+                                         NULL /* read_prefs */,
+                                         opts,
+                                         reply_ptr,
+                                         error)) {
+      GOTO (fail);
+   }
+   ok = true;
```

Comment:
```suggestion
   ok = mongoc_client_command_with_opts (collection->client,
                                         collection->db,
                                         &cmd,
                                         NULL /* read_prefs */,
                                         opts,
                                         reply_ptr,
                                         error);

```
Minor simplification.